### PR TITLE
Build wheels with all the dependencies included

### PR DESCRIPTION
**Pull request by angeloskath**
Updates the CI to add wheel building jobs that include the dependencies in order to make a PyPI package.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ repos:
     hooks:
     -   id: clang-format
         exclude: '^(lib/bxzstr|cmake|lib/stb).*|.c$'
--   repo: https://github.com/psf/black
-    rev: 22.10.0
+-   repo: https://github.com/psf/black-pre-commit-mirror
+    rev: 23.12.1
```
Comment (awni) [inline .pre-commit-config.yaml:8]: 👍

Comment (awni) [reply .pre-commit-config.yaml:8]: Maybe we add the isort here as well just to be consistent accross the board:

```
-   repo: https://github.com/pycqa/isort
    rev: 5.12.0
    hooks:
    -   id: isort
        args:
            - --profile=black
```

**Review by awni — APPROVED** [review]
Wow - this is some real CI foo 👊

**Review by andresy — APPROVED** [review]
LGTM!

**Pull request merged**